### PR TITLE
KIALI-1445 KIALI-1446 KIALI-1447 Create a template for documentation, Menu and Overview section

### PR DESCRIPTION
**Pull request by aljesusg**
Still working 
result

### CODE REVIEW

**Comment by aljesusg** [comment]
I sent this only for feedback, still working
@theute Deploy/netlify throw an error that in local not happen , it seems taht we need to add something to work with asciidocs in the build I need to research about
```
12:16:05 PM: ERROR 2018/08/31 10:16:05 asciidoctor / asciidoc not found in $PATH: Please install.
12:16:05 PM:                   Leaving AsciiDoc content unrendered.
12:16:05 PM: ERROR 2018/08/31 10:16:05 asciidoctor / asciidoc not found in $PATH: Please install.
12:16:05 PM:                   Leaving AsciiDoc content unrendered.
```

On the other hand, I don't like the blink effect everytime that you change the page, I don't know what do you think but I am gonna try to avoid this.

**Comment by aljesusg** [comment]
Solved the problem asciidoctor with sudo dnf install asciidoctor or gem install asciidoctor if we are working with ruby for Hugo

**Comment by aljesusg** [comment]
Fixed @theute added a Gemfile with the gem ^^

**Comment by theute** [comment]
Note that Kiali.io is generated from Travis (to Amazon) so you likely need
to install asciidoc from there as well
https://github.com/kiali/kiali.io/blob/master/.travis.yml

**Comment by aljesusg** [comment]
Ok, is there a Ruby version in Amazon instance? If not, what Operating System is ?

File: config.toml [hunk 1]
```diff
@@ -4,29 +4,67 @@ title = "Kiali"
...
+      "https://use.fontawesome.com/releases/v5.3.1/css/all.css"
+    ]
 
 [menu]
   [[menu.main]]
-  identifier = "swagger"
-  name = "API"
-  url = "/api/overview"
-  weight = 40
+      identifier = "swagger"
```
Comment (abonas) [inline config.toml]: why are all the diffs in this file needed? some are only tabs, some are comment out of part of the content (they might be, I just want to make sure)

File: content/documentation/overview/index.adoc [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+date: 2018-06-20T19:04:38+02:00
+draft: false
+type: "documentation"
+---
+
+= Kiali Overview
+:sectnums:
+:toc: left
+toc::[]
+:toc-title: Overview Contents
```
Comment (abonas) [inline content/documentation/overview/index.adoc]: I think it should always  be "Content", not in plural

File: config.toml (hunk 1 above)
Comment (aljesusg) [reply config.toml]: Sorry I resolved the review :'(  Additional css for asccidocs icons, tabulation format to be more readable and I added the full documentation with comments to test the Index for documentation and let to the next editors the menu ready.

File: content/documentation/overview/index.adoc (hunk 2 above)
Comment (aljesusg) [reply content/documentation/overview/index.adoc]: Thanks, I need help to this document, I don't know exactly if you wanna a resume of all the Application or a full description of each page and some review of the writing will be great maybe Julie can help here.

File: .gitignore [hunk 3]
```diff
@@ -1,5 +1,6 @@
 output
 public
+Gemfile.lock
```
Comment (abonas) [inline .gitignore]: why gitignore the lock? we are only using a gem here, not writing one, iirc it should be commited, but I might have forgot the best practices :)

Comment (aljesusg) [reply .gitignore]: I used to upload it until I saw that they did not do it in miq ^^. Anyway the version is set. What do you think ? Should I upload the lock?

**Comment by theute** [comment]
If you hide/remove the title 3 about "customers" (and have abonas ack) then we could merge and iterate from there.

Let's not discuss the content too much here, this is mainly about the documentation build itself.

Thanks !

**Comment by aljesusg** [comment]
Updated !!! Thanks @abonas @theute

**Pull request merged**